attribute-mixins/validation.js

```diff
@@ -0,0 +1,97 @@
+import { result } from 'lodash';
+
+class Validator {
+  static required(value) {
+    return value !== undefined;
+  }
+}
+
+const ValidationMixin = (superclass) => class extends superclass {
+  static validatorClass() { return Validator; }
+
+  constructor({ value, validations, autoValidate }) {
+    super({ value: value });
+
+    buildValidation.call(this, validations);
+
+    if (autoValidate !== false) {
+      this.on('change', () => {
+        this.validate();
+      });
+      this.validate();
+    }
+  }
+
+  ValidateOneDry(validationKey) {
+    const validationValue = result(this.validations, validationKey);
+    const validator = this.constructor.validatorClass();
+
+    if (validator[validationKey]) {
+      return validator[validationKey](this.value, validationValue);
+    }
+
+    return true;
+  }
+
+  validateAllDry() {
+    const errors = {};
+    const validationKeys = this.validations ? Object.keys(this.validations) : [];
+
+    if (this.validations && validationKeys.length) {
+      const required = this.ValidateOneDry('required');
+      if (!required) {
+        if (this.validations.required) errors.required = required;
+      } else {
+        const keysToValidate = validationKeys.filter((key) => {
+          return key !== 'required';
+        });
+
+        keysToValidate.forEach((key) => {
+          const validateResult = this.ValidateOneDry(key);
+          if (validateResult !== true) {
+            errors[key] = validateResult;
+          }
+        });
+      }
+    }
+
+    return errors;
+  }
+};
+
+export default ValidationMixin;
```

Comment:
it's weird that we have mixin validation that export a ValidationMixin and in attribute-mixins/validation.js we return a ValidationMixin. I think we should rename this class to AttributeValidationMixin

Reply:
When you import it you can name it like you want. Still i will rename it